snekbox/nsjail.py

```diff
@@ -102,19 +98,27 @@ def _create_dynamic_cgroups(self) -> str:
         pids.mkdir(parents=True, exist_ok=True)
         mem.mkdir(parents=True, exist_ok=True)
 
+        # Cgroups v2 renamed the limit file in their new API
+        if self.cgroups_version == 2:
+            limit_name = "max"
+            swap_name = "swap"
+        else:
+            limit_name = "limit_in_bytes"
+            swap_name = "memsw"
+
         # Swap limit cannot be set to a value lower than memory.limit_in_bytes.
         # Therefore, this must be set before the swap limit.
         #
         # Since child cgroups are dynamically created, the swap limit has to be set on the parent
         # instead so that children inherit it. Given the swap's dependency on the memory limit,
         # the memory limit must also be set on the parent. NsJail only sets the memory limit for
         # child cgroups, not the parent.
-        (mem / "memory.limit_in_bytes").write_text(mem_max, encoding="utf-8")
+        (mem / f"memory.{limit_name}").write_text(mem_max, encoding="utf-8")
 
         try:
             # Swap limit is specified as the sum of the memory and swap limits.
             # Therefore, setting it equal to the memory limit effectively disables swapping.
```

Comment:
This description is for v1's behaviour. Does `memory.swap.max` in v2 also behave the same way?